File: WorldFeed/Server/src/WorldFeed.Controllers/Trends/TrendsController.cs
namespace WorldFeed.Controllers.Trends
{
    using System.Threading.Tasks;

    using WorldFeed.Common.Public.Models.Interfaces;
    using WorldFeed.Common.Public.Parameters.TrendsClient;
    using WorldFeed.Common.Web;
    using WorldFeed.Core.Controllers;

    public class TrendsController : ITrendsController
    {
        private readonly ITrendsQueryExecutor trendsQueryExecutor;

        public TrendsController(ITrendsQueryExecutor trendsQueryExecutor)
        {
            this.trendsQueryExecutor = trendsQueryExecutor;
        }


        public Task<ITwitterResult<IGetTrendsAtResult[]>> GetPlaceTrendsAtAsync(IGetTrendsAtParameters parameters, ITwitterRequest request)
        {
            return this.trendsQueryExecutor.GetPlaceTrendsAtAsync(parameters, request);
        }

        public Task<ITwitterResult<ITrendLocation[]>> GetTrendLocationsAsync(IGetTrendsLocationParameters parameters, ITwitterRequest request)
        {
            return this.trendsQueryExecutor.GetTrendLocationsAsync(parameters, request);
        }

        public Task<ITwitterResult<ITrendLocation[]>> GetTrendsLocationCloseToAsync(IGetTrendsLocationCloseToParameters parameters, ITwitterRequest request)
        {
            return this.trendsQueryExecutor.GetTrendsLocationCloseToAsync(parameters, request);
        }
    }
}

File: Sharebook/Client/src/app/controllers/Trends/TrendsController.ts
import {Inject, Injectable} from "@angular/core";

import {ITrendsController} from "../../core/Core/Controllers/ITrendsController";
import {ITwitterResult} from "../../core/Core/Web/TwitterResult";
import {ITwitterRequest} from "../../core/Public/Models/Interfaces/ITwitterRequest";
import {ITrendsQueryExecutor, ITrendsQueryExecutorToken} from "./TrendsQueryExecutor";
import {IGetTrendsAtParameters} from "../../core/Public/Parameters/TrendsClient/GetTrendsAtParameters";
import {IGetTrendsLocationParameters} from "../../core/Public/Parameters/TrendsClient/GetTrendsLocationParameters";
import {IGetTrendsLocationCloseToParameters} from "../../core/Public/Parameters/TrendsClient/GetTrendsLocationCloseToParameters";
import {IGetTrendsAtResult} from "../../core/Public/Models/Interfaces/IGetTrendsAtResult";
import {ITrendLocation} from "../../core/Public/Models/Interfaces/ITrendLocation";

@Injectable({
  providedIn: 'root',
})
export class TrendsController implements ITrendsController {
  private readonly _trendsQueryExecutor: ITrendsQueryExecutor;

  constructor(@Inject(ITrendsQueryExecutorToken) trendsQueryExecutor: ITrendsQueryExecutor) {
    this._trendsQueryExecutor = trendsQueryExecutor;
  }

  public getPlaceTrendsAtAsync(parameters: IGetTrendsAtParameters, request: ITwitterRequest): Promise<ITwitterResult<IGetTrendsAtResult[]>> {
    return this._trendsQueryExecutor.getPlaceTrendsAtAsync(parameters, request);
  }

  public getTrendLocationsAsync(parameters: IGetTrendsLocationParameters, request: ITwitterRequest): Promise<ITwitterResult<ITrendLocation[]>> {
    return this._trendsQueryExecutor.getTrendLocationsAsync(parameters, request);
  }

  public getTrendsLocationCloseToAsync(parameters: IGetTrendsLocationCloseToParameters, request: ITwitterRequest): Promise<ITwitterResult<ITrendLocation[]>> {
    return this._trendsQueryExecutor.getTrendsLocationCloseToAsync(parameters, request);
  }
}
